See bug https://bugs.gentoo.org/836723

The error is this:

/var/tmp/portage/net-misc/turbovnc-2.2.7-r2/work/turbovnc-2.2.7/unix/Xvnc/programs/Xserver/os/access.c: In function 'GetLocalClientCreds':
/var/tmp/portage/net-misc/turbovnc-2.2.7-r2/work/turbovnc-2.2.7/unix/Xvnc/programs/Xserver/os/access.c:1178:18: error: storage size of 'peercred' isn't known
 1178 |     struct ucred peercred;
      |                  ^~~~~~~~

--- ./unix/Xvnc/programs/Xserver/os/CMakeLists.txt~	2022-06-12 15:41:05.980270914 +0300
+++ ./unix/Xvnc/programs/Xserver/os/CMakeLists.txt	2022-06-12 15:43:14.113823213 +0300
@@ -1,7 +1,7 @@
 include_directories(../render ../../../lib/libsha1)
 
 add_definitions(-DUNIXCONN -DTCPCONN -DHASXDMAUTH ${ServerOSDefines} -DXDMCP
-	-DHAVE_SHA1_IN_LIBSHA1)
+	-DHAVE_SHA1_IN_LIBSHA1 -D_GNU_SOURCE)
 if(CMAKE_SYSTEM_NAME STREQUAL "SunOS")
 	add_definitions(-DLOCALCONN)
 elseif(NOT CMAKE_SYSTEM_NAME STREQUAL "Linux")
